spec/consensus/reactor/reactor.md

```diff
@@ -0,0 +1,355 @@
+# Consensus Reactor
+
+Tendermint Core is a state machine replication framework and part of the stack used in the Cosmos ecosystem to build distributed applications.
+Tendermint Core uses its northbound API, ABCI, to communicate with applications.
+South of Tendermint Core, is the OS' network stack.
+
+Tendermint Core is implemented in a modular way, separating protocol implementations in **reactors**.
+Reactors communicate in with their counterparts on other nodes using the P2P layer, through what we will call the **P2P-I**.
+
+
+```
+                                                      SDK Apps
+                                                   ==================
+ Applications                                        Cosmos SDK    
+====================================ABCI=============================
+  Mempool Reactor     Evidence Reactor    Consensus Reactor   PEX ...
+- - - - - - - - - - - - - - - P2P-I - - - - - - - - - - - - - - - - - 
+                                 P2P
+=====================================================================
+                            Network Stack
+```
+
+This document focuses on the interactions between the Consensus Reactor and the P2P layer.
+The Consensus reactor is, itself, divided into two layers, State and Communication. 
+
+The **State** layer keeps the state and transition functions described in the [The latest gossip on BFT consensus](https://arxiv.org/abs/1807.0493).
+
+The **Communication** layer (AKA, Gossip) handles messages received through the P2P layer, triggering updates of peer information and, when conditions are met, of the consensus state.
+The communication layer also reacts to events in the consensus state and, periodically, propagates information on the node's state to its neighbors. 
```

Comment:
"events in the consensus state": wording is a bit unclear

Reply:
I was influenced by the use of pub-sub to send messages. Abstracted and rephrased.